e2e/fixtures/pages/ExploreProfilesPage.ts

```diff
@@ -10,28 +10,32 @@ export class ExploreProfilesPage extends PyroscopePage {
     super(page, '/a/grafana-pyroscope-app/profiles-explorer', urlParams.toString());
   }
 
-  goto(explorationType: string | undefined) {
-    if (!explorationType) {
-      return super.goto();
-    }
+  goto(explorationType: ExplorationType, urlSearchParams = new URLSearchParams()) {
+    const urlParams = new URLSearchParams({
+      ...Object.fromEntries(DEFAULT_EXPLORE_PROFILES_URL_PARAMS),
+      ...Object.fromEntries(urlSearchParams),
+    });
 
-    const urlParams = new URLSearchParams(DEFAULT_EXPLORE_PROFILES_URL_PARAMS);
     urlParams.set('explorationType', explorationType);
 
     return super.goto(urlParams.toString());
   }
 
+  /* Data source */
```

Comment:
In a future PR, we should split the responsibilities among several, smaller classes.